spotify_project/artists/tests/test_views.py

```diff
@@ -2,36 +2,59 @@
 
 from artists.factories import ArtistFactory
 from artists.models import Artist
+from artists.views import all_artists, single_artist
+from django.test import Client, RequestFactory
 from django.urls import reverse
+from libraries.factories import LibraryFactory
 from testing import BaseTestCase
 from tracks.factories import TrackFactory
+from users.factories import CustomUserFactory
 
 
 class AllArtistsTestCase(BaseTestCase):
     url = reverse('artists:all_artists')
 
     def setUp(self):
         super().setUp()
-        ArtistFactory.create_batch(3)
+        request_factory = RequestFactory()
```

Comment:
There is a lot of mess going on with tests in this PR. Looks like ran into issues with getting the user logged in. I can help with that.

Reply:
Let's forget whatever those tests were the first time. I have redone them much better.